packages/block-library/src/video/edit.js

```diff
@@ -97,9 +97,15 @@ function VideoEdit( {
 			setAttributes( { src: undefined, id: undefined } );
 			return;
 		}
+
 		// sets the block's attribute and updates the edit component from the
 		// selected media
-		setAttributes( { src: media.url, id: media.id } );
+		setAttributes( {
+			src: media.url,
+			id: media.id,
+			poster:
+				media.image?.src !== media.icon ? media.image?.src : undefined,
```

Comment:
This emulates what seems to be used in core in a couple of places:
https://github.com/WordPress/wordpress-develop/blob/b83b01e1b01429dc4a6da49ec23e1f7dfc4ae446/src/wp-includes/media-template.php#L386